test/lint.js

```diff
@@ -38,162 +35,135 @@ const argv = yargs
   .alias('help', '?')
   .parse(process.argv.slice(2));
 
+/** @type {object} */
+const spinner = ora({
+  stream: process.stdout,
+});
+
 /**
- * @param {string[]} files
- * @return {boolean}
+ * Recursively load one or more files and/or directories passed as arguments and check for any errors.
+ *
+ * @param {string[]} files The files to test
+ * @returns {object} Errors by relative file path.
  */
-function load(...files) {
-  return files.reduce((prevHasErrors, file) => {
+const checkFiles = (...files) => {
+  let errors = {};
+  for (let file of files) {
     if (file.indexOf(__dirname) !== 0) {
       file = path.resolve(__dirname, '..', file);
     }
```

Comment:
It would probably be a good idea to map the whole files array upfront:

```
files = files.map(file => file.startsWith(__dirname) ? file : path.resolve(__dirname, '..', file));
```

PS: The arrow function could be extracted to a `resolveFile()` helper function.